scipy/_lib/doccer.py

```diff
@@ -1,26 +1,39 @@
-''' Utilities to allow inserting docstring fragments for common
-parameters into function and method docstrings'''
+"""Utilities to allow inserting docstring fragments for common
+parameters into function and method docstrings."""
 
+from collections.abc import Callable, Iterable
+from typing import Any, ParamSpec, TypeAlias, TypeVar
 import sys
 
 __all__ = [
-    'docformat', 'inherit_docstring_from', 'indentcount_lines',
-    'filldoc', 'unindent_dict', 'unindent_string', 'extend_notes_in_docstring',
-    'replace_notes_in_docstring', 'doc_replace'
+    "docformat",
+    "inherit_docstring_from",
+    "indentcount_lines",
+    "filldoc",
+    "unindent_dict",
+    "unindent_string",
+    "extend_notes_in_docstring",
+    "replace_notes_in_docstring",
+    "doc_replace",
 ]
 
+P = ParamSpec("P")
+R = TypeVar("R")
```

Comment:
Unless you use these typevar-likes in other modules too (which I wouldn't recommend), then the convention is to prefix their names with a `"_'"`, i.e.

```suggestion
_P = ParamSpec("_P")
_R = TypeVar("_R")
```